File: ui/src/models/pet.ts
import { defaultOng, Ong } from './ong'

export interface Pet {
  id: number
  type: string
  breed: string
  color: string
  size: string
  sex: 'F' | 'M'
  personality?: string
  wasAdopted: boolean
  photo: string
  ongId: number
  ong: Ong
}

export const defaultPet = (): Pet => {
  return {
    id: 0,
    type: '',
    breed: '',
    color: '',
    size: '',
    sex: 'F',
    personality: '',
    wasAdopted: false,
    photo: '',
    ongId: 0,
    ong: defaultOng(),
  }
}

File: api/Models/Pet.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace apifmu.Models
{
    public class Pet
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Breed { get; set; }

        [Required]
        public string Color { get; set; }

        [Required]
        public string Size { get; set; }

        [Required]
        public char Sex { get; set; }

        public string Personality { get; set; }

        [Required]
        public bool WasAdopted { get; set; }

        [Required]
        public string Photo { get; set; }

        [ForeignKey("Ong")]
        public int OngId { get; set; }

        public Ong Ong { get; set; }

        [ForeignKey("User")]
        public int? UserId { get; set; }

        public User User { get; set; }
    }
}
